routes/noteRoutes.js

```diff
@@ -0,0 +1,110 @@
+const express = require("express");
+const mongoose = require("mongoose");
+const Note = require("../models/Note");
+const router = express.Router();
+
+
+router.get("/", async (req, res) => {
+    try {
+        const page = parseInt(req.query.page) || 1;  
+        const limit = parseInt(req.query.limit) || 10;  
+        const skip = (page - 1) * limit;
+
+        const totalNotes = await Note.countDocuments();  
+        const notes = await Note.find().skip(skip).limit(limit);
+
+        if (notes.length === 0) {
+            return res.status(404).json({ 
+                success: false, 
+                message: "No notes found" 
+            });
+        }
+
+        res.status(200).json({
+            success: true,
+            currentPage: page,
+            totalPages: Math.ceil(totalNotes / limit),
+            totalNotes,
+            notes
+        });
+    } catch (error) {
+        res.status(500).json({ success: false, message: "Server error!", error: error.message });
+    }
+});
+
+router.post("/", async (req, res) => {
+    try {
+        const { title, content } = req.body;
+        if (!title || !content) {
+            return res.status(400).json({ message: "All Fields are Required" })
+        }
+
+        const newNode = new Note({ title, content })
+        await newNode.save()
+        res.status(201).json(newNode)
+    } catch (error) {
+        res.status(500).json({ message: "server error!", error: error.message })
```

Comment:
Keep consistent error message:

```suggestion
        res.status(500).json({ message: "Server error!", error: error.message });
```